code/modules/power/supermatter/supermatter.dm

```diff
@@ -678,6 +678,22 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		if(produces_gas)
 			env.merge(removed)
 			air_update_turf()
+	if(takes_damage)
+		for(var/t in RANGE_TURFS(1, loc))
+			if(!isspaceturf(t))
+				continue
+			var/turf/turf_to_check = t
+			if(LAZYLEN(turf_to_check.atmos_adjacent_turfs))
+				var/integrity = get_integrity()
+				if(integrity < 10)
+					damage += max((power * 0.0005) * DAMAGE_INCREASE_MULTIPLIER, 0)
+				else if(integrity < 25)
+					damage += max((power * 0.0009) * DAMAGE_INCREASE_MULTIPLIER, 0)
+				else if(integrity < 45)
+					damage += max((power * 0.005) * DAMAGE_INCREASE_MULTIPLIER, 0)
+				else if(integrity < 75)
+					damage += max((power * 0.002) * DAMAGE_INCREASE_MULTIPLIER, 0)
```

Comment:
Cap these by MAX_DAMAGE or some joker is going to set energy to some absurd number with fusion-heated CO2 and instantly detonate the crystal.

Reply:
putting this in the damage creating block would prob be a good idea.